### PR TITLE
NAS-117350 / 22.12 / Make WebUI compatible with iSCSI API changes

### PR DESCRIPTION
**Pull request by bvasilenko**
**Testing**

Go to **Sharing > Block (iSCSI) Shares Targets > Settings**, and follow the ticket's description

### CODE REVIEW

File: src/app/pages/sharing/iscsi/target-global-configuration/target-global-configuration.component.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ export class TargetGlobalConfigurationComponent implements OnInit {
     basename: ['', Validators.required],
     isns_servers: [[] as string[]],
     pool_avail_threshold: [null as number],
+    listen_port: [null as number],
```
Comment (undsoft) [inline src/app/pages/sharing/iscsi/target-global-configuration/target-global-configuration.component.ts:37]: Needs required validator and [required] attribute.

File: src/app/pages/sharing/iscsi/portal/portal-form/portal-form.component.html [hunk 2]
```diff
@@ -31,23 +31,14 @@
...
-                  type="number"
-                  [required]="true"
-                ></ix-input>
-              </div>
+              <ix-select
+                [formControlName]="'ip' + listPrefix + i"
+                [label]="labels.ip | translate"
+                [tooltip]="tooltips.ip | translate"
+                [options]="listenOptions$"
+                [multiple]="true"
```
Comment (undsoft) [inline src/app/pages/sharing/iscsi/portal/portal-form/portal-form.component.html:39]: I don't think we need multiple=true anymore.

File: src/app/pages/sharing/iscsi/iscsi-wizard/iscsi-wizard.component.ts [hunk 3]
```diff
@@ -626,7 +626,7 @@ export class IscsiWizardComponent implements WizardConfiguration {
     this.iscsiService.listPortals().pipe(untilDestroyed(this)).subscribe((portals) => {
       const field = _.find(this.wizardConfig[1].fieldConfig, { name: 'portal' }) as FormSelectConfig;
       for (const portal of portals) {
-        const ips = portal.listen.map((ip) => ip.ip + ':' + ip.port);
+        const ips = portal.listen.map((ip) => ip.ip);
```
Comment (undsoft) [inline src/app/pages/sharing/iscsi/iscsi-wizard/iscsi-wizard.component.ts:629]: Port not removed from the form that is shown when you select `Create New` under Portals.

File: src/app/pages/sharing/iscsi/iscsi-wizard/iscsi-wizard.component.ts [hunk 4]
```diff
@@ -626,7 +626,7 @@ export class IscsiWizardComponent implements WizardConfiguration {
...
         field.options.push({ label: portal.tag + ' (' + ips + ')', value: portal.id });
```
Comment (undsoft) [inline src/app/pages/sharing/iscsi/iscsi-wizard/iscsi-wizard.component.ts:630]: Authorized networks not removed from Wizard.

**Review by undsoft — CHANGES_REQUESTED** [review]

**Comment by bvasilenko** [comment]
All the requested changes now addressed

File: src/app/pages/sharing/iscsi/iscsi-wizard/iscsi-wizard.component.ts (hunk 3 above)
Comment (undsoft) [reply src/app/pages/sharing/iscsi/iscsi-wizard/iscsi-wizard.component.ts]: Not properly addressed. There is still a port field.

**Review by undsoft — APPROVED** [review]
Wizard wasn't updated correctly, otherwise okay.

**Comment by bvasilenko** [comment]
Now wizard is working as well

**Pull request merged**